src/_pytest/mark/structures.py

```diff
@@ -65,17 +66,27 @@ def get_empty_parameterset_mark(
     return mark
 
 
+# Singleton type for HIDDEN_PARAM, as described in:
+# https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions
+class _HiddenParam(enum.Enum):
+    token = 0
```

Comment:
An arbitrary integer feels wrong here. Why can't it be a new object?
```suggestion
    token = object()
```

Reply:
Ack. I didn't realize that enum required integers. Makes sense then. Marking this thread as resolved.